exporter/datadogexporter/trace_connection.go

```diff
@@ -45,19 +46,18 @@ type traceEdgeConnectionImpl struct {
 }
 
 const (
-	traceEdgeTimeout       time.Duration = 10 * time.Second
 	traceEdgeRetryInterval time.Duration = 10 * time.Second
 )
 
 // createTraceEdgeConnection returns a new traceEdgeConnection
-func createTraceEdgeConnection(rootURL, apiKey string, buildInfo component.BuildInfo) traceEdgeConnection {
+func createTraceEdgeConnection(rootURL, apiKey string, buildInfo component.BuildInfo, settings exporterhelper.TimeoutSettings) traceEdgeConnection {
 
 	return &traceEdgeConnectionImpl{
 		traceURL:  rootURL + "/api/v0.2/traces",
 		statsURL:  rootURL + "/api/v0.2/stats",
 		buildInfo: buildInfo,
 		apiKey:    apiKey,
-		client:    utils.NewHTTPClient(traceEdgeTimeout),
+		client:    utils.NewHTTPClient(settings),
```

Comment:
This changes the timeout value from 10 seconds to 15 seconds (if I read https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/77d48a50e0b6f46dfea3159e8ea375abc5e3deb5/exporter/datadogexporter/factory.go#L49-L53 correctly), is that fine? Should we change `defaulttimeoutSettings` to return 10 seconds instead?

Reply:
Yes, this is fine (we care about the timeout of the batch processor for traces but that's about it, the timeout here should be fine no matter the value)